ksql-parser/src/main/java/io/confluent/ksql/parser/properties/with/WithClauseProperties.java

```diff
@@ -117,4 +119,42 @@ private static void throwOnUnknownProperty(
           + String.join(",", onlyInProvided));
     }
   }
+
+  public Optional<Character> getValueDelimiter() {
+    final String providedValueDelimiter = getString(CommonCreateConfigs.VALUE_DELIMITER_PROPERTY);
+    if (providedValueDelimiter != null) {
```

Comment:
nit: move `if (providedValueDelimiter == null) { return Optional.empty(); }` to the beginning of the function.